crates/adapters/src/integrated/delta_table/input.rs

```diff
@@ -210,9 +217,186 @@ impl DeltaTableInputEndpointInner {
         }
     }
 
+    /// Load the entire table snapshot as a single "select * where <filter>" query.
+    async fn read_unordered_snapshot(
+        &self,
+        input_stream: &mut dyn ArrowStream,
+        receiver: &mut Receiver<PipelineState>,
+    ) {
+        // Execute the snapshot query; push snapshot data to the circuit.
+        info!(
+            "delta_table {}: reading initial snapshot",
+            &self.endpoint_name,
+        );
+
+        let mut snapshot_query = "select * from snapshot".to_string();
+        if let Some(filter) = &self.config.snapshot_filter {
+            snapshot_query = format!("{snapshot_query} where {filter}");
+        }
+
+        self.execute_snapshot_query(&snapshot_query, "initial snapshot", input_stream, receiver)
+            .await;
+
+        //let _ = self.datafusion.deregister_table("snapshot");
+        info!(
+            "delta_table {}: finished reading initial snapshot",
+            &self.endpoint_name,
+        );
+    }
+
+    /// Load the initial snapshot by issuing a sequence of queries for monotonically
+    /// increasing timestamp ranges.
+    async fn read_ordered_snapshot(
+        &self,
+        input_stream: &mut dyn ArrowStream,
+        schema: &Relation,
+        receiver: &mut Receiver<PipelineState>,
+    ) {
+        self.read_ordered_snapshot_inner(input_stream, schema, receiver)
+            .await
+            .unwrap_or_else(|e| self.consumer.error(true, e));
+    }
+
+    async fn read_ordered_snapshot_inner(
+        &self,
+        input_stream: &mut dyn ArrowStream,
+        schema: &Relation,
+        receiver: &mut Receiver<PipelineState>,
+    ) -> Result<(), AnyError> {
+        let timestamp_column = self.config.timestamp_column.as_ref().unwrap();
+
+        let timestamp_field = schema.field(timestamp_column).unwrap();
+
+        // The following unwraps are safe, as validated in `validate_timestamp_column`.
+        let lateness = timestamp_field.lateness.as_ref().unwrap();
+
+        // Query the table for min and max values of the timestamp column that satisfy the filter.
+        let bounds_query =
```

Comment:
is this efficient if the table has one of these properties you described?

Reply:
Hopefully.  If it's too expensive, then reading the ranges will likely be expensive too.